src/rules/media-query-value-dollar-variable/index.js

```diff
@@ -0,0 +1,62 @@
+import { utils } from "stylelint"
+import { namespace } from "../../utils"
+
+export const ruleName = namespace("media-query-value-dollar-variable")
+
+export const messages = utils.ruleMessages(ruleName, {
+  rejected: "Unexpected dollar-variable as a value in a media query",
+  expected: "Expected a dollar-variable (e.g. $var) to be used as a value in a media query",
+})
+
+export default function (expectation, options) {
+  return function (root, result) {
+    const validOptions = utils.validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [ "always", "never" ],
+    }, {
+      actual: options,
+      optional: true,
+      possible: {
+        ignore: [ "calculations", "funcitons" ],
```

Comment:
Yeah, I wasn't sure whether to use options up till the last moment (even after pushing). Now since we 100% agree on how this rule should work I can purge them.